### ISSUE TITLE
Argument is deprecated

### ISSUE THREAD

**Issue by marianafalcho** [terraform-aws-modules/terraform-aws-s3-bucket#142]
Before you submit an issue, please perform the following first:

1. Remove the local `.terraform` directory (! ONLY if state is stored remotely, which hopefully you are following that best practice!): `rm -rf .terraform/` _Done_
2. Re-initialize the project root to pull down modules: `terraform init`_Done_
3. Re-attempt your terraform plan or apply and check if the issue still persists _Done_

## Versions

- Module version [Required]:
```
version = "2.14.1"
```
- Terraform version:
```
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.75.0
```
- Provider version(s):
```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.69"
    }
  }
}
```

## Reproduction Code [Required]
This is my code
```
module "s3_bucket" {
  source  = "terraform-aws-modules/s3-bucket/aws"
  version = "2.14.1"
  bucket  = "${data.aws_iam_account_alias.current.account_alias}-results"

  block_public_acls       = true
  block_public_policy     = true
  ignore_public_acls      = true
  restrict_public_buckets = true
  lifecycle_rule = [
    {
      id      = "canary_wip_results_cleanup"
      enabled = true
      prefix  = "canary_wip/"
      transition = [
        {
          days          = 15
          storage_class = "STANDARD_IA"
        }
      ]
      expiration = {
        days = 30
      }
    }
  ]

  server_side_encryption_configuration = {
    rule = {
      apply_server_side_encryption_by_default = {
        sse_algorithm = "AES256"
      }
    }
  }
}

```

 No
 Yes

1. Created code
2. Ran `terraform init`. I did this twice to troubleshoot and ensure the provider was properly installed
3. Ran `terraform plan`. Got error:
```
│ Warning: Argument is deprecated
│ 
│   with module.s3_bucket.aws_s3_bucket.this,
│   on .terraform/modules/s3_bucket/main.tf line 5, in resource "aws_s3_bucket" "this":
│    5: resource "aws_s3_bucket" "this" {
│ 
│ Use the aws_s3_bucket_website_configuration resource instead
│ 
│ (and 27 more similar warnings elsewhere)
```

## Expected behavior

No warning to be shown.

## Actual behavior

Warning:
```
│ Warning: Argument is deprecated
│ 
│   with module.cloudwatch_synthetic_canaries_prod_s3_bucket.aws_s3_bucket.this,
│   on .terraform/modules/cloudwatch_synthetic_canaries_prod_s3_bucket/main.tf line 5, in resource "aws_s3_bucket" "this":
│    5: resource "aws_s3_bucket" "this" {
│ 
│ Use the aws_s3_bucket_website_configuration resource instead
│ 
│ (and 27 more similar warnings elsewhere)
```

## Additional context

Just trying to create 2 buckets using this module and I get the warning listed above.

**Comment by antonbabenko** [comment]
Duplicate of #138 . Will be fixed in #139 in the next major release (hopefully during today).

For now, you can lock the version of the AWS provider to be less than 3.75:

```
version = "~> 3.65, < 3.75"
```

**Comment by brent-moffit** [comment]
I'm still seeing this warning using module version `3.0.1` and aws provider version `4.9.0` on terraform `1.1.7`.  I've run both `terraform init -upgrade` and deleted the `.terraform` folder for a clean `init` but still see the warning when I run `terraform validate` or `plan`.  Is there an additional fix needed for provider version `4.9.0`?
